cdisc_rules_engine/utilities/dataset_preprocessor.py

```diff
@@ -118,7 +120,16 @@ def _merge_datasets(
         )
 
         # merge datasets based on their type
-        if self._rule_processor.is_relationship_dataset(right_dataset_domain_name):
+        if right_dataset_domain_name == "RELREC":
```

Comment:
Are we properly detecting this domain?

Reply:
This is comparing the value in:
```yaml
Match Datasets
- Name:
```
to "RELREC"

What did you have in mind?

I did notice that "domain_name" is not being added to the dataset metadata for datasets that do not have a "DOMAIN" column. Technically these aren't domains and they shouldn't have this populated, but a lot of the code depends on this metadata being populated for these datasets because the dataset name is not being used.  I updated this in the latest commit. In the future, we will want to fix this everywhere so that datasets and domains are distinguishes properly. It will become more apparent in other standards like adam.